spec/unit/configurer_spec.rb

```diff
@@ -599,6 +600,22 @@ def method_missing(*args)
         expect(@agent.retrieve_catalog({})).to eq(@catalog)
       end
 
+      it "should not make a node request or pluginsync when a cached catalog is successfully retrieved" do
+        Puppet::Node.indirection.expects(:find).never
+        @agent.expects(:download_plugins).never
+
+        @agent.expects(:prepare_and_retrieve_catalog_from_cache).returns(@catalog)
```

Comment:
Could you set the expectation on `Puppet::Resource::Catalog.indirection.find` with `ignore_terminus => true` as opposed to the private method of the class we're testing?